### PR TITLE
Use Selective Omissions

### PR DESCRIPTION
**Pull request by scbedd**
1. Bump `doc-warden` version
2. Leverage new feature to ignore some readmes
3. Stop treating said readmes as exceptions.

### CODE REVIEW

File: eng/.docsettings.yml [hunk 1]
```diff
@@ -9,8 +9,14 @@ omitted_paths:
   - samples/*
   - common/SmokeTests/*
   - sdk/*/samples/*
+  - sdk/*/*/samples/*
   - sdk/*/swagger/*
   - doc/ApiDocGeneration/*
+  - doc/*
```
Comment (weshaggard) [inline eng/.docsettings.yml]: What is the difference between `doc/*` and `doc/` now?

File: eng/.docsettings.yml [hunk 2]
```diff
@@ -9,8 +9,14 @@ omitted_paths:
   - samples/*
   - common/SmokeTests/*
   - sdk/*/samples/*
+  - sdk/*/*/samples/*
```
Comment (weshaggard) [inline eng/.docsettings.yml]: What places do we have this structure for samples?

Comment (scbedd) [reply eng/.docsettings.yml]: Honestly, I'm not certain what the one ABOVE it is for. 

The one that's there currently matches something like:
```
sdk/eventhubs/samples
sdk/    *    /samples
```
Mine will match 
```

sdk/eventhubs/Azure.Messaging.Eventhubs/samples
sdk/    *    /             *           /samples
```

Which I feel like omits a bunch more samples than the one that they have.

File: eng/.docsettings.yml (hunk 1 above)
Comment (scbedd) [reply eng/.docsettings.yml]: Good point. I'm not certain how I didn't see something directly above it.

File: eng/.docsettings.yml [hunk 3]
```diff
@@ -8,9 +8,13 @@ omitted_paths:
   - sdk/*/*.Management.*/*
   - samples/*
   - common/SmokeTests/*
-  - sdk/*/samples/*
+  - sdk/*/*/samples/*
   - sdk/*/swagger/*
-  - doc/ApiDocGeneration/*
+  - doc/*
+  - sdk/search/
+  - sdk/keyvault/
```
Comment (weshaggard) [inline eng/.docsettings.yml]: Shouldn't these paths still be associated with a bug as they should each have a README in their root folder.

Comment (scbedd) [reply eng/.docsettings.yml]: I actually deliberately removed them from the buglist when I added them to the omission list. If they should be associated with the bug, I'll add the items back. 

I am adding them to the omission list deliberately based on the discussion that caused this issue to be opened in the first place :P

**Comment by scbedd** [comment]
Closing pending discussion in Azure/azure-sdk-tools#42.

Need to establish across the board whether or not these service level _should_ be enforced.

**Pull request closed without merging**